checker/match_test.go

```diff
@@ -118,6 +147,31 @@ func TestMatchHTTPNil(t *testing.T) {
 	Expect(matchHTTP(nil, req)).To(BeTrue())
 }
 
+// Test HTTPPaths panic on invalid data.
+func TestPanicHTTPPaths(t *testing.T) {
+	RegisterTestingT(t)
+	paths := []*proto.HTTPMatch_PathMatch{{&proto.HTTPMatch_PathMatch_Exact{Exact: "/foo"}}}
+	Expect(func() {
+		matchHTTPPaths(paths, "foo")
+	}).Should(Panic())
+
+	// Check type of panic also.
+	Expect(func() {
+		defer func() {
+			if r := recover(); r != nil {
+				if _, ok := r.(*InvalidDataFromDataPlane); ok {
+					// do nothing
+				} else {
+					panic(r)
+				}
+			} else {
+				panic(r)
+			}
+		}()
+		matchHTTPPaths(paths, "foo")
+	}).ShouldNot(Panic())
```

Comment:
This second test clause subsumes the first one, so I don't think you need both.  Also, I think this would be simpler to read if you wrote the `Expect()` clause directly against the result of `recover()`  E.g.

```
defer func() {
    Expect(recover()).To(BeAssignableToTypeOf(&InvalidDataFromDataPlane{}))
}()
matchHTTPPaths(paths, "foo")
```
You shouldn't then have to wrap the whole thing in an `Expect()` clause.

Reply:
good suggestion! Thanks.
Do u think it's ok to merge as the protobuf changes have not yet gone into Felix? I know protobuf's are forward compatible so it should be ok but still!

Reply:
Fine by me to merge, but take heed of my note about numbering!